src/validation/parsePermissions.ts

```diff
@@ -21,211 +21,212 @@ import {
   PermissionIdentityType,
 } from '@coveo/platform-client';
 
-export const processPermissionList = (
-  caseInsensitiveDoc: CaseInsensitiveDocument<PrimitivesValues>,
-  documentBuilder: DocumentBuilder,
-  documentPath: PathLike
-) => {
-  ensurePermissionArray(caseInsensitiveDoc, documentPath);
-  new KnownKeys<Document['permissions']>(
-    'permissions',
-    caseInsensitiveDoc
-  ).whenExists((permissions) => {
-    permissions!.forEach(
-      (permission: Recordable<PermissionSet> | Recordable<PermissionLevel>) => {
-        const caseInsensitivePermission =
-          new CaseInsensitiveDocument<PrimitivesValues>(permission);
-
-        new KnownKeys('permissionsets', caseInsensitivePermission)
-          .whenExists<PermissionLevel>((permissionLevel) =>
-            processPermissionLevel(
-              permissionLevel,
-              documentBuilder,
-              documentPath
+class PermissionParser {
```

Comment:
Could be done in another PR, but I see we write a lot of code for validations.
Would it be possible to simplify this by leveraging a library such as zod?

Reply:
Probably, but I guess bueno is doing the same thing here.

Reply:
Whatever the lib, is just that we could probably reduce the codebase if we use one of them.

Reply:
Sorry. I meant, the parser already leverages bueno.